Issue with installation on Debian

Hello,

I tried to set up MapServer version 8.6.0 on a Debian machine, but I encountered the following error:


```
/home/sboomi/wms-server/MapServer/src/maprasterquery.c: In function ‘msRASTERLayerGetShape’:
/home/sboomi/wms-server/MapServer/src/maprasterquery.c:1199:14: error: ‘GDT_Int8’ undeclared (first use in this function); did you mean ‘GDT_Int64’?
 1199 |         case GDT_Int8:
      |              ^~~~~~~~
      |              GDT_Int64
/home/sboomi/wms-server/MapServer/src/maprasterquery.c:1199:14: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [CMakeFiles/mapserver.dir/build.make:1038: CMakeFiles/mapserver.dir/src/maprasterquery.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:227: CMakeFiles/mapserver.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

<details>
<summary>Stacktrace details</summary>


```
[  1%] Generating renderers/mvt/vector_tile.pb-c.c, renderers/mvt/vector_tile.pb-c.h
[  1%] Building C object CMakeFiles/mapserver.dir/renderers/mvt/vector_tile.pb-c.c.o
[  2%] Building C object CMakeFiles/mapserver.dir/src/fontcache.c.o
[  3%] Building C object CMakeFiles/mapserver.dir/src/cgiutil.c.o
[  3%] Building C object CMakeFiles/mapserver.dir/src/mapgeos.c.o
[  4%] Building C object CMakeFiles/mapserver.dir/src/maporaclespatial.c.o
[  4%] Building C object CMakeFiles/mapserver.dir/src/mapsearch.c.o
[  5%] Building CXX object CMakeFiles/mapserver.dir/src/mapwms.cpp.o
[  5%] Building C object CMakeFiles/mapserver.dir/src/classobject.c.o
[  6%] Building C object CMakeFiles/mapserver.dir/src/mapgml.c.o
[  7%] Building C object CMakeFiles/mapserver.dir/src/mapoutput.c.o
[  7%] Building C object CMakeFiles/mapserver.dir/src/mapwmslayer.c.o
[  8%] Building C object CMakeFiles/mapserver.dir/src/layerobject.c.o
[  8%] Building C object CMakeFiles/mapserver.dir/src/mapgraticule.c.o
[  9%] Building C object CMakeFiles/mapserver.dir/src/mapows.c.o
[  9%] Building CXX object CMakeFiles/mapserver.dir/src/mapogcapi.cpp.o
[ 10%] Building C object CMakeFiles/mapserver.dir/src/mapservutil.c.o
[ 11%] Building C object CMakeFiles/mapserver.dir/src/mapxbase.c.o
[ 11%] Building C object CMakeFiles/mapserver.dir/src/maphash.c.o
[ 12%] Building C object CMakeFiles/mapserver.dir/src/mapowscommon.c.o
[ 12%] Building C object CMakeFiles/mapserver.dir/src/mapshape.c.o
[ 13%] Building C object CMakeFiles/mapserver.dir/src/mapxml.c.o
[ 13%] Building C object CMakeFiles/mapserver.dir/src/mapbits.c.o
[ 14%] Building C object CMakeFiles/mapserver.dir/src/maphttp.c.o
[ 15%] Building C object CMakeFiles/mapserver.dir/src/mapparser.c.o
[ 15%] Building CXX object CMakeFiles/mapserver.dir/src/mapstring.cpp.o
[ 16%] Building C object CMakeFiles/mapserver.dir/src/mapxmp.c.o
[ 16%] Building C object CMakeFiles/mapserver.dir/src/mapcairo.c.o
[ 17%] Building C object CMakeFiles/mapserver.dir/src/mapimageio.c.o
[ 17%] Building C object CMakeFiles/mapserver.dir/src/mappluginlayer.c.o
[ 18%] Building C object CMakeFiles/mapserver.dir/src/mapsymbol.c.o
[ 19%] Building C object CMakeFiles/mapserver.dir/src/mapchart.c.o
[ 19%] Building C object CMakeFiles/mapserver.dir/src/mapimagemap.c.o
[ 20%] Building C object CMakeFiles/mapserver.dir/src/mappool.c.o
[ 20%] Building C object CMakeFiles/mapserver.dir/src/maptclutf.c.o
[ 21%] Building C object CMakeFiles/mapserver.dir/src/mapcluster.c.o
[ 21%] Building C object CMakeFiles/mapserver.dir/src/mapio.c.o
[ 22%] Building CXX object CMakeFiles/mapserver.dir/src/mappostgis.cpp.o
[ 23%] Building C object CMakeFiles/mapserver.dir/src/maptemplate.c.o
[ 23%] Building C object CMakeFiles/mapserver.dir/src/mapcontext.c.o
[ 24%] Building C object CMakeFiles/mapserver.dir/src/mapjoin.c.o
[ 24%] Building C object CMakeFiles/mapserver.dir/src/mappostgresql.c.o
[ 25%] Building C object CMakeFiles/mapserver.dir/src/mapthread.c.o
[ 25%] Building C object CMakeFiles/mapserver.dir/src/mapcopy.c.o
[ 26%] Building C object CMakeFiles/mapserver.dir/src/maplabel.c.o
[ 27%] Building CXX object CMakeFiles/mapserver.dir/src/mapprimitive.cpp.o
[ 27%] Building C object CMakeFiles/mapserver.dir/src/maptile.c.o
[ 28%] Building C object CMakeFiles/mapserver.dir/src/mapcpl.c.o
[ 28%] Building C object CMakeFiles/mapserver.dir/src/maplayer.c.o
[ 29%] Building C object CMakeFiles/mapserver.dir/src/mapproject.c.o
[ 29%] Building C object CMakeFiles/mapserver.dir/src/maptime.c.o
[ 30%] Building C object CMakeFiles/mapserver.dir/src/mapcrypto.c.o
[ 31%] Building C object CMakeFiles/mapserver.dir/src/maplegend.c.o
[ 31%] Building C object CMakeFiles/mapserver.dir/src/hittest.c.o
[ 32%] Building C object CMakeFiles/mapserver.dir/src/maptree.c.o
[ 32%] Building C object CMakeFiles/mapserver.dir/src/mapdebug.c.o
[ 33%] Building C object CMakeFiles/mapserver.dir/src/maplexer.c.o
[ 33%] Building C object CMakeFiles/mapserver.dir/src/mapquantization.c.o
[ 34%] Building CXX object CMakeFiles/mapserver.dir/src/mapunion.cpp.o
[ 35%] Building C object CMakeFiles/mapserver.dir/src/mapdraw.c.o
[ 35%] Building C object CMakeFiles/mapserver.dir/src/maplibxml2.c.o
[ 36%] Building CXX object CMakeFiles/mapserver.dir/src/mapquery.cpp.o
[ 36%] Building C object CMakeFiles/mapserver.dir/src/maputil.c.o
[ 37%] Building C object CMakeFiles/mapserver.dir/src/strptime.c.o
[ 37%] Building C object CMakeFiles/mapserver.dir/src/mapdrawgdal.c.o
[ 38%] Building C object CMakeFiles/mapserver.dir/src/mapraster.c.o
[ 39%] Building CXX object CMakeFiles/mapserver.dir/src/mapuvraster.cpp.o
[ 39%] Building C object CMakeFiles/mapserver.dir/src/mapdummyrenderer.c.o
[ 40%] Building C object CMakeFiles/mapserver.dir/src/mapobject.c.o
[ 40%] Building C object CMakeFiles/mapserver.dir/src/maprasterquery.c.o
/home/sboomi/wms-server/MapServer/src/maprasterquery.c: In function ‘msRASTERLayerGetShape’:
/home/sboomi/wms-server/MapServer/src/maprasterquery.c:1199:14: error: ‘GDT_Int8’ undeclared (first use in this function); did you mean ‘GDT_Int64’?
 1199 |         case GDT_Int8:
      |              ^~~~~~~~
      |              GDT_Int64
/home/sboomi/wms-server/MapServer/src/maprasterquery.c:1199:14: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [CMakeFiles/mapserver.dir/build.make:1038: CMakeFiles/mapserver.dir/src/maprasterquery.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:227: CMakeFiles/mapserver.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

``` 

</details>


# Specs 

**MapServer version:**

8.6.0

**Debian version:**

`Linux version 6.1.0-41-amd64 (debian-kernel@lists.debian.org) (gcc-12 (Debian 12.2.0-14+deb12u1) 12.2.0, GNU ld (GNU Binutils for Debian) 2.40) #1 SMP PREEMPT_DYNAMIC Debian 6.1.158-1 (2025-11-09)`

**GDAL version**

* GDAL: GDAL 3.12.0 "Chicoutimi", released 2018/99/99 (debug build)
* `gdalinfo`: GDAL 3.12.0 "Chicoutimi", released 2018/99/99 (debug build)

> [!NOTE]
> This version of GDAL has been compiled manually since I thought the problem was due to an ancient version of GDAL from APT.

**CMake version**


```
cmake version 4.2.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```


**Results of the `configure.out.txt` file:**

<details>
  <summary>configure.out.txt</summary>


```txt
-- Requiring C++17
-- Requiring C++17 - done
-- Requiring C17
-- Requiring C17 - done
-- Found Proj 9.1
-- Found GDAL 3.6
-- CMake Version: 4.2
-- Build Type: 
-- Python MapScript output directory: /home/sboomi/wms-server/mapserver-build/src/mapscript/python
-- Python Executable: /home/sboomi/.local/bin/python3.14
-- Python Version: 3.14.2
-- Python site packages: /home/sboomi/.local/share/uv/python/cpython-3.14.2-linux-x86_64-gnu/lib/python3.14/site-packages
-- * Summary of configured options for this build
--  * Mandatory components
--   * GDAL: /home/sboomi/.local/lib/libgdal.so
--   * PROJ: /usr/lib/x86_64-linux-gnu/libproj.so
--   * png: /usr/lib/x86_64-linux-gnu/libpng.so
--   * jpeg: /usr/lib/x86_64-linux-gnu/libjpeg.so
--   * freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so
--  * Optional components
--   * PCRE2: /usr/lib/x86_64-linux-gnu/libpcre2-posix.so
--   * GIF: /usr/lib/x86_64-linux-gnu/libgif.so
--   * MYSQL: disabled
--   * FRIBIDI: /usr/lib/x86_64-linux-gnu/libfribidi.so
--   * HARFBUZZ: /usr/lib/x86_64-linux-gnu/libharfbuzz.so
--   * CAIRO: /usr/lib/x86_64-linux-gnu/libcairo.so
--   * SVGCAIRO: disabled
--   * RSVG: disabled
--   * CURL: /usr/lib/x86_64-linux-gnu/libcurl.so
--   * PIXMAN: disabled
--   * LIBXML2: /usr/lib/x86_64-linux-gnu/libxml2.so
--   * POSTGIS: /usr/lib/x86_64-linux-gnu/libpq.so
--   * GEOS: /usr/lib/x86_64-linux-gnu/libgeos_c.so
--   * FastCGI: /usr/lib/x86_64-linux-gnu/libfcgi.so
--   * PROTOBUFC: /usr/lib/x86_64-linux-gnu/libprotobuf-c.so
--   * Oracle Spatial: disabled
--   * Exempi XMP: disabled
--  * Optional features
--   * WMS SERVER: ENABLED
--   * WFS SERVER: ENABLED
--   * WCS SERVER: ENABLED
--   * SOS SERVER: ENABLED
--   * OGCAPI SERVER: ENABLED
--   * Install HTML Bootstrap resources for OGCAPI: disabled
--   * WMS CLIENT: ENABLED
--   * WFS CLIENT: ENABLED
--   * ICONV: ENABLED
--   * Thread-safety support: disabled
--   * KML output: disabled
--   * XML Mapfile support: disabled
--  * Mapscripts
--   * Python: ENABLED
--   * PHPNG: disabled
--   * PERL: ENABLED
--   * RUBY: disabled
--   * JAVA: ENABLED
--   * C#: disabled
--   * V8 Javascript: disabled
--   * Apache Module (Experimental): disabled
--   * Python MapScript Annotations: disabled
-- 
-- PROJECT_BINARY_DIR is set to /home/sboomi/wms-server/mapserver-build
-- Will install files to /opt
-- Will install libraries to /opt/lib
-- Configuring done (0.4s)
-- Generating done (0.1s)
-- Build files have been written to: /home/sboomi/wms-server/mapserver-build

```

</details>

# Steps to reproduce

```sh
$ git clone 
$ git checkout rel-8-6-0
$ cd ..
$ mkdir mapserver-build
$ cd mapserver-build
$ cmake -DCMAKE_INSTALL_PREFIX=/opt  \  
 -DWITH_CLIENT_WFS=ON  \
 -DWITH_CLIENT_WMS=ON \    
 -DWITH_CURL=ON  \  
 -DWITH_SOS=ON  \ 
 -DWITH_PHP=ON  \   
 -DWITH_PERL=ON  \   
 -DWITH_RUBY=OFF   \   
 -DWITH_JAVA=ON   \     
 -DWITH_CSHARP=OFF  \   
 -DWITH_PYTHON=ON    \ 
 -DWITH_SVGCAIRO=OFF  \
 -DWITH_ORACLESPATIAL=OFF   \  
-DWITH_MSSQL2008=OFF  \
 ../MapServer >../configure.out.txt
$ make
```

The error  looks like it's GDAL-related. What should I do? Downgrade and rebuild GDAL? Thank you in advance for your response.

Regards,

--- a/src/maprasterquery.c
+++ b/src/maprasterquery.c
@@ -1269,7 +1269,10 @@ int msRASTERLayerGetShape(layerObj *layer, shapeObj *shape, resultObj *record) {
 
         switch (rlinfo->eDataType) {
         case GDT_Byte:
+#if GDAL_VERSION_MAJOR > 3 ||                                                  \
+    (GDAL_VERSION_MAJOR == 3 && GDAL_VERSION_MINOR >= 7)
         case GDT_Int8:
+#endif
         case GDT_UInt16:
         case GDT_Int16:
         case GDT_Int32:

